posix/syscall.c

```diff
@@ -0,0 +1,605 @@
+
+            ret = oe_link(oldpath, newpath);
+            goto done;
+        }
+        case OE_SYS_unlink:
+        {
+            const char* pathname = (const char*)arg1;
+
+            ret = oe_unlink(pathname);
+            goto done;
+        }
+        case OE_SYS_rename:
+        {
+            const char* oldpath = (const char*)arg1;
+            const char* newpath = (const char*)arg2;
+
+            ret = oe_rename(oldpath, newpath);
+            goto done;
+        }
+        case OE_SYS_truncate:
+        {
+            const char* path = (const char*)arg1;
+            ssize_t length = (ssize_t)arg2;
+
+            ret = oe_truncate(path, length);
+            goto done;
+        }
+        case OE_SYS_mkdir:
+        {
+            const char* pathname = (const char*)arg1;
+            uint32_t mode = (uint32_t)arg2;
+
+            ret = oe_mkdir(pathname, mode);
+            goto done;
+        }
+        case OE_SYS_rmdir:
+        {
+            const char* pathname = (const char*)arg1;
+            ret = oe_rmdir(pathname);
+            goto done;
+        }
+        case OE_SYS_access:
+        {
+            const char* pathname = (const char*)arg1;
+            int mode = (int)arg2;
+
+            ret = oe_access(pathname, mode);
+            goto done;
+        }
+        case OE_SYS_getdents:
+        case OE_SYS_getdents64:
+        {
+            unsigned int fd = (unsigned int)arg1;
+            struct oe_dirent* ent = (struct oe_dirent*)arg2;
+            unsigned int count = (unsigned int)arg3;
+            ret = oe_getdents(fd, ent, count);
+            goto done;
+        }
+        case OE_SYS_ioctl:
+        {
+            int fd = (int)arg1;
+            unsigned long request = (unsigned long)arg2;
+            long p1 = arg3;
+            long p2 = arg4;
+            long p3 = arg5;
+            long p4 = arg6;
+
+            ret = oe_ioctl(fd, request, p1, p2, p3, p4);
+            goto done;
+        }
+        case OE_SYS_fcntl:
+        {
+            int fd = (int)arg1;
+            int cmd = (int)arg2;
+            int arg = (int)arg3;
+            ret = oe_fcntl(fd, cmd, arg);
+            goto done;
+        }
+        case OE_SYS_mount:
+        {
+            const char* source = (const char*)arg1;
+            const char* target = (const char*)arg2;
+            const char* fstype = (const char*)arg3;
+            unsigned long flags = (unsigned long)arg4;
+            void* data = (void*)arg5;
+
+            ret = oe_mount(source, target, fstype, flags, data);
+            goto done;
+        }
+        case OE_SYS_umount2:
+        {
+            const char* target = (const char*)arg1;
+            int flags = (int)arg2;
+
+            (void)flags;
+
+            ret = oe_umount(target);
+            goto done;
+        }
+        case OE_SYS_getcwd:
+        {
+            char* buf = (char*)arg1;
+            size_t size = (size_t)arg2;
+
+            if (!oe_getcwd(buf, size))
```

Comment:
No because the **getcwd()** function and the **getcwd** are different. The function returns a pointer to the first argument (char*), but the system call by that name return long. Here's the MUSL implementation:

```
char *getcwd(char *buf, size_t size)
{
        char tmp[buf ? 1 : PATH_MAX];
        if (!buf) {
                buf = tmp;
                size = sizeof tmp;
        } else if (!size) {
                errno = EINVAL;
                return 0;
        }
        long ret = syscall(SYS_getcwd, buf, size);
        if (ret < 0)
                return 0;
        if (ret == 0 || buf[0] != '/') {
                errno = ENOENT;
                return 0;
        }
        return buf == tmp ? strdup(buf) : buf;
}
```